cpp/src/arrow/flight/sql/odbc/tests/connection_info_test.cc

```diff
@@ -0,0 +1,1453 @@
+  this->connect();
+
+  validate(conn, SQL_MAX_CURSOR_NAME_LEN, static_cast<SQLUSMALLINT>(0));
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_MAX_IDENTIFIER_LEN) {
+  this->connect();
+
+  validate(conn, SQL_MAX_IDENTIFIER_LEN, static_cast<SQLUSMALLINT>(65535));
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_MAX_INDEX_SIZE) {
+  this->connect();
+
+  validate(conn, SQL_MAX_INDEX_SIZE, static_cast<SQLUINTEGER>(0));
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_MAX_PROCEDURE_NAME_LEN) {
+  this->connect();
+
+  validate(conn, SQL_MAX_PROCEDURE_NAME_LEN, static_cast<SQLUSMALLINT>(0));
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_MAX_ROW_SIZE) {
+  this->connect();
+
+  validate(conn, SQL_MAX_ROW_SIZE, L"");
+
+  this->disconnect();
+}
+
+TEST_F(FlightSQLODBCMockTestBase, Test_SQL_MAX_ROW_SIZE_INCLUDES_LONG) {
+  this->connect();
+
+  validate(conn, SQL_MAX_ROW_SIZE_INCLUDES_LONG, L"N");
+
+  this->disconnect();
+}
+
+TEST_F(FlightSQLODBCMockTestBase, Test_SQL_MAX_SCHEMA_NAME_LEN) {
+  this->connect();
+
+  validate(conn, SQL_MAX_SCHEMA_NAME_LEN, static_cast<SQLUSMALLINT>(0));
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_MAX_STATEMENT_LEN) {
+  this->connect();
+
+  validate(conn, SQL_MAX_STATEMENT_LEN, static_cast<SQLUINTEGER>(0));
+
+  this->disconnect();
+}
+
+TEST_F(FlightSQLODBCMockTestBase, Test_SQL_MAX_TABLE_NAME_LEN) {
+  this->connect();
+
+  validate(conn, SQL_MAX_TABLE_NAME_LEN, static_cast<SQLUSMALLINT>(0));
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_MAX_TABLES_IN_SELECT) {
+  this->connect();
+
+  validate(conn, SQL_MAX_TABLES_IN_SELECT, static_cast<SQLUSMALLINT>(0));
+
+  this->disconnect();
+}
+
+TEST_F(FlightSQLODBCMockTestBase, Test_SQL_MAX_USER_NAME_LEN) {
+  this->connect();
+
+  validate(conn, SQL_MAX_USER_NAME_LEN, static_cast<SQLUSMALLINT>(0));
+
+  this->disconnect();
+}
+
+// Scalar Function Information
+
+TEST_F(FlightSQLODBCMockTestBase, Test_SQL_CONVERT_FUNCTIONS) {
+  this->connect();
+
+  validate(conn, SQL_CONVERT_FUNCTIONS, static_cast<SQLUINTEGER>(0));
+
+  this->disconnect();
+}
+
+TEST_F(FlightSQLODBCMockTestBase, Test_SQL_NUMERIC_FUNCTIONS) {
+  this->connect();
+
+  validate(conn, SQL_NUMERIC_FUNCTIONS, static_cast<SQLUINTEGER>(4058942));
+
+  this->disconnect();
+}
+
+TEST_F(FlightSQLODBCMockTestBase, Test_SQL_STRING_FUNCTIONS) {
+  this->connect();
+
+  validate(conn, SQL_STRING_FUNCTIONS, static_cast<SQLUINTEGER>(1304));
+
+  this->disconnect();
+}
+
+TEST_F(FlightSQLODBCMockTestBase, Test_SQL_SYSTEM_FUNCTIONS) {
+  this->connect();
+
+  validate(conn, SQL_SYSTEM_FUNCTIONS, static_cast<SQLUINTEGER>(5));
+
+  this->disconnect();
+}
+
+TYPED_TEST(FlightSQLODBCTestBase, Test_SQL_TIMEDATE_ADD_INTERVALS) {
+  this->connect();
+
+  validate(conn, SQL_TIMEDATE_ADD_INTERVALS, static_cast<SQLUINTEGER>(511));
```

Comment:
Suggest to test against default values provided by the driver.

```suggestion
  validate(conn, SQL_TIMEDATE_ADD_INTERVALS, static_cast<SQLUINTEGER>(SQL_FN_TSI_FRAC_SECOND | SQL_FN_TSI_SECOND | SQL_FN_TSI_MINUTE | SQL_FN_TSI_HOUR |
      SQL_FN_TSI_DAY | SQL_FN_TSI_WEEK | SQL_FN_TSI_MONTH | SQL_FN_TSI_QUARTER |
      SQL_FN_TSI_YEAR));
```